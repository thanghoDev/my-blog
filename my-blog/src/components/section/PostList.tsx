import React from 'react';
import Image from 'next/future/image';
import Link from 'next/link';
import { Col } from 'react-bootstrap';

// types
import { Blog } from '@/types/blog';

type PostListProps = {
  data: Blog[];
  title: string;
};

function PostList({ data, title }: PostListProps) {
  return (
    <Col md={6} className='me-4'>
      <Link href={`category${title}`}>
        <h2 className='mb-5 text-decoration-underline text-capitalize fs-5 postTitle'>
          <a>{title}</a>
        </h2>
      </Link>

      {data.map((item) => (
        <div key={item.id} className='d-flex mb-3'>
          <Image width='165' height='180' src={item.images} alt={item.alt} />
          <div className='ms-4'>
            <Link href={`detail/${item.id}`}>
              <h2 className='fs-6 text-black text-capitalize'>
                <a>{item.title}</a>
              </h2>
            </Link>
            <p className='description'>{item.description}</p>
            <p className='text-capitalize category'>
              Dave Rogers <code className='text-secondary'>in</code>{' '}
              {item.category}
            </p>
          </div>
        </div>
      ))}
    </Col>
  );
}

export default PostList;
