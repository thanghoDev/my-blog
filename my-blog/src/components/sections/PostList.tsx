import React from 'react';
import Link from 'next/link';
import { Col } from 'react-bootstrap';

// constants
import { CATEGORY } from '@/constants/pages';

// types
import { Blog } from '@/types/blog';
import Posts from '../card/Posts';

type PostListProps = {
  data: Blog[];
  title: string;
};

function PostList({ data, title }: PostListProps) {
  return (
    <Col md={6} className='me-4'>
      <Link href={`/${CATEGORY}/${title}`}>
        <h2 className='mb-5 text-decoration-underline text-capitalize fs-5 postTitle'>
          {title}
        </h2>
      </Link>

      {data.map((item) => (
        <Posts key={item.id} posts={item} />
      ))}
    </Col>
  );
}

export default PostList;
