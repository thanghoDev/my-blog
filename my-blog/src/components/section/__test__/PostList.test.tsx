import React from 'react';
import { render, screen } from '@testing-library/react';

import PostList from '../PostList';

describe('testing Politics section', () => {
  const politics = [
    {
      id: 1,
      title: 'title',
      description: 'description',
      category: 'PostList',
      view: 150,
      images: 'https://i.ibb.co/10hj3h1/PostList.jpg',
      alt: 'PostList',
      content: 'content',
    },
  ];
  const HeadingTitle = 'headingTitle';
  test('render section', () => {
    const { container } = render(
      <PostList title={HeadingTitle} data={politics} />
    );
    expect(container.firstChild).toMatchSnapshot();
    const heading = screen.getByRole('heading', {
      name: /headingTitle/i,
    });
    expect(heading).toBeInTheDocument();

    const image = screen.getByRole('img', {
      name: /PostList/i,
    });
    expect(image).toBeInTheDocument();

    const title = screen.getByRole('heading', {
      name: 'title',
    });
    expect(title).toBeInTheDocument();
  });
});
