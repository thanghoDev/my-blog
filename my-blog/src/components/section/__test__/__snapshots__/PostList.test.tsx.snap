// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing PostList section render section 1`] = `
<div
  class="me-4 col-md-6"
>
  <h2
    class="mb-5 text-decoration-underline text-capitalize fs-5 postTitle"
  >
    <a>
      headingTitle
    </a>
  </h2>
  <div
    class="d-flex mb-3"
  >
    <img
      alt="postList"
      data-nimg="future"
      decoding="async"
      height="180"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2FPostList.jpg&w=384&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2FPostList.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2F10hj3h1%2FPostList.jpg&w=384&q=75 2x"
      style="color: transparent;"
      width="165"
    />
    <div
      class="ms-4"
    >
      <a
        class="fs-6 text-decoration-none text-black text-capitalize postTitle"
        href="/detail/1"
      >
        title
      </a>
      <p
        class="description"
      >
        description
      </p>
      <p
        class="text-capitalize category"
      >
        Dave Rogers 
        <code
          class="text-secondary"
        >
          in
        </code>
         
        <a
          class="text-black text-decoration-none"
          href="/category/postList"
        >
          postList
        </a>
      </p>
    </div>
  </div>
</div>
`;
